src/twisted/internet/test/test_unix.py

```diff
@@ -383,26 +383,34 @@ def test_fileDescriptorOverrun(self):
         test_fileDescriptorOverrun.skip = sendmsgSkip
 
 
-    def test_multiFileDescriptorReceivedPerRecvmsg(self):
+    def _SendmsgMixinFileDescriptorReceivedDriver(self, ancillaryPacker):
         """
-        _SendmsgMixin handles multiple file descriptors per recvmsg, calling
-        L{IFileDescriptorReceiver.fileDescriptorReceived} once per received
-        file descriptor.
+        Drive _SendmsgMixin via sendmsg socket calls to check that
+        L{IFileDescriptorReceiver.fileDescriptorReceived} is called once
+        for each file descriptor received in the ancillary messages.
+        ancillaryPacker is called with a list of two file descriptors and
+        should return a two-tuple where: the first item is an iterable of
+        zero or more (cmsg_level, cmsg_type, cmsg_data) tuples for actual
+        sending via sendmsg; the second item is an integer indicating the
+        expected number of FDs to be received.
         """
         # Strategy:
         # - Create a UNIX socketpair.
         # - Associate one end to a FakeReceiver and FakeProtocol.
-        # - Call sendmsg on the other end with two FDs as ancillary data.
+        # - Call sendmsg on the other end to send FDs as ancillary data.
+        #   (ancillary data is obtained calling ancillaryPacker with
+        #   the two FDs associated with the socketpair)
         # - Call doRead in the FakeReceiver.
         # - Verify results on FakeProtocol.
 
         # TODO: replace FakeReceiver test approach with one based in
         # IReactorSocket.adoptStreamConnection once AF_UNIX support is
         # implemented; see https://twistedmatrix.com/trac/ticket/5573.
 
-        from socket import socketpair
+        import socket
```

Comment:
IIRC, these imports are here because they will otherwise generate import failures on Windows, maybe I'm wrong. All these tests should be skipped on Windows, given that they're testing UNIX sockets, but I'm not sure they are, as of now.

PS: That does not apply to the "import socket" here, which I put in place to handle the socket.error exception in the "dynamic" test skipping code you pointed out below, which I will improve; but it does apply, for example, to the "from socket import socketpair".